cylc/flow/workflow_events.py

```diff
@@ -302,14 +305,26 @@ def _send_mail(
             env=env,
             stdin_str=message
         )
-        if self.proc_pool.closed:
-            # Run command in foreground if process pool is closed
-            self.proc_pool.run_command(proc_ctx)
-            self._run_event_handlers_callback(proc_ctx)
+        self._run_cmd(proc_ctx, callback=self._run_event_mail_callback)
+
+    def _run_cmd(self, ctx, callback):
+        """Queue or directly run a command and its callback.
+
+        Queue the command to the subprocess pool if possible, or otherwise
+        run it in the foreground but subject to the subprocess pool timeout.
+
+        """
+        if not self.proc_pool.closed:
+            # Queue it to the subprocess pool.
+            self.proc_pool.put_command(ctx, callback=callback)
         else:
-            # Run command using process pool otherwise
-            self.proc_pool.put_command(
-                proc_ctx, callback=self._run_event_mail_callback)
+            # Run it in the foreground, but use the subprocess pool timeout.
+            try:
+                self.proc_pool.run_command(ctx, float(self.proc_timeout))
+            except TimeoutExpired:
+                ctx.ret_code = -9
```

Comment:
The `timeout` command exits `124` if the timeout is exceeded. If setting an exit code, we might as well match this for consistency:

```suggestion
                ctx.ret_code = 124
```

Reply:
I initially took this suggestion, but I've backed out of it on implementing the process kill in response to your main comment.

The return code `-9` from `subprocess.wait()` (which is how I'm getting it now, not hardwiring it) reflects the value of the signal used to kill the process (SIGKILL is 9). Is that OK with you?